tests/virt_autotest/libvirt_virtual_network_init.pm

```diff
@@ -45,31 +45,30 @@ sub run_test {
     virt_autotest::virtual_network_utils::enable_libvirt_log();
 
     #VM HOST SSH SETUP
-    virt_autotest::virtual_network_utils::ssh_setup();
+    virt_autotest::virtual_network_utils::ssh_setup() if (!(get_var('VIRT_AUTOTEST')));
```

Comment:
Why this?

Reply:
After finished setup_dns_service, there was created testing SSH KEY as /var/testvirt.net/.ssh/id_rsa.pub and copy this key to guest system, thus all VIRT_AUTOTEST will let vm host access guest system without password. 
As the same, there was not required  ssh_setup() and ssh-copy-id step for all VIRT_AUTOTEST.
used with` if (!(get_var('VIRT_AUTOTEST')));` here, just work for QAM environment.

Reply:
If you really need to store your SSH key in `/var/testvirt.net/.ssh/` then please modify the `virt_autotest::virtual_network_utils::ssh_setup()` and avoid creating the key elsewhere.

```suggestion
    virt_autotest::virtual_network_utils::ssh_setup();
```

Reply:
> So, we will use virt_autotest::virtual_network_utils::ssh_setup(); together, no copy ssh key to guest for QA part.

Perfect 👍  Then you should probably remove this condition and also upgrade the `ssh_setup()` as you mentioned:
```perl
sub ssh_setup {
  my $default_ssh_key = (!(get_var('VIRT_AUTOTEST'))) ? "/root/.ssh/id_rsa.pub" : "/var/testvirt.net/.ssh/id_rsa.pub";
  if (script_run("[[ -f $default_ssh_key ]]") != 0) {
    assert_script_run "ssh-keygen -t rsa -P '' -f $default_ssh_key";
  }
}
```